dbxio/volume/volume_commands.py

```diff
@@ -166,97 +180,101 @@ def download_volume(
-    if volume_info.volume_type is None:
-        raise ValueError(f'Volume {catalog_name}.{schema_name}.{volume_name} does not exist')
 
-    volume = Volume(
-        catalog=catalog_name,
-        schema=schema_name,
-        name=volume_name,
-        volume_type=volume_info.volume_type,
-        storage_location=volume_info.storage_location,
-    )
+    volume = Volume.from_url(get_volume_url(catalog_name, schema_name, volume_name, volume_path), client)
     logger.info(f'Downloading volume: {volume.full_name}')
     logger.debug(f'Volume type: {volume.volume_type}')
 
-    if volume_info.volume_type == VolumeType.EXTERNAL:
-        assert volume_info.storage_location, f'External volume must have a storage location, got {volume_info=}'
-        _download_external_volume(local_path=path, storage_location=volume_info.storage_location)
-    elif volume_info.volume_type == VolumeType.MANAGED:
+    if volume.volume_type == VolumeType.EXTERNAL:
+        assert volume.storage_location, f'External volume must have a storage location, got {volume=}'
+        _download_external_volume(
+            local_path=path,
+            storage_location=str(volume.storage_location),
+            volume_path=volume.path,
+        )
+    elif volume.volume_type == VolumeType.MANAGED:
         _download_managed_volume(local_path=path, volume=volume, client=client)
     else:
-        raise ValueError(f'Unknown volume type: {volume_info.volume_type}')
+        raise ValueError(f'Unknown volume type: {volume.volume_type}')
 
     logger.info(f'Volume {volume.full_name} was successfully downloaded to {path}')
 
-    return path
+    return path / volume.path
 
 
 def _write_external_volume(
     path: Path,
     catalog_name: str,
     schema_name: str,
     volume_name: str,
+    volume_path: str,
     client: 'DbxIOClient',
     max_concurrency: int,
     create_volume_if_not_exists: bool,
     force: bool,
 ):
-    catalog = client.workspace_api.catalogs.get(catalog_name)
-    catalog_properties = catalog.properties if catalog.properties else {}
-
-    default_external_location = catalog_properties.get(DEFAULT_EXTERNAL_LOCATION)
-    if not default_external_location:
-        raise ValueError(
-            'Catalog does not have a default external location (aka container in storage account). '
-            'Ask data engineers for creation of a default external location for your catalog.'
-        )
-    storage_url = client.workspace_api.external_locations.get(default_external_location).url
-    assert storage_url, f'External location {default_external_location} does not have a URL'
-    object_storage_client = ObjectStorageClient.from_url(storage_url)
+    volume_exists = _exists_volume(catalog_name, schema_name, volume_name, client)
+    if volume_exists:
+        volume = Volume.from_url(get_volume_url(catalog_name, schema_name, volume_name), client=client)
+        assert volume.storage_location, f'External volume must have a storage location, got {volume=}'
```

Comment:
Can we do this check in `Volume.from_url` factory?

Reply:
This is for mypy check